test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs

```diff
@@ -0,0 +1,866 @@
+                WarningsAsErrors = true,
+                Optimize = true,
+                DelaySign = true,
+                PublicSign = true,
+                EmitEntryPoint = true,
+                GenerateXmlDocumentation = true,
+                PreserveCompilationContext = true
+            };
+        }
+
+        [Fact]
+        public void It_does_not_throw_when_the_project_json_is_empty()
+        {
+            var json = new JObject();
+            Action action = () => GetProject(json);
+
+            action.ShouldNotThrow<Exception>();
+        }
+
+        [Fact]
+        public void It_sets_Name_to_the_passed_ProjectName_if_one_is_not_set_in_the_ProjectJson()
+        {
+            _emptyProject.Name.Should().Be(ProjectName);
+        }
+
+        [Fact]
+        public void It_sets_Name_to_the_Name_in_the_ProjectJson_when_one_is_set()
+        {
+            const string nameInprojectJson = "some name in the project.json";
+            var json = new JObject();
+            json.Add("name", nameInprojectJson);
+            var project = GetProject(json);
+
+            project.Name.Should().Be(nameInprojectJson);
+        }
+
+        [Fact]
+        public void It_sets_the_project_file_path()
+        {
+            _emptyProject.ProjectFilePath.Should().Be(ProjectFilePath);
+        }
+
+        [Fact]
+        public void It_sets_the_version_to_one_when_it_is_not_set()
+        {
+            _emptyProject.Version.Should().Be(new NuGetVersion("1.0.0"));
+        }
+
+        [Fact]
+        public void It_sets_the_version_to_the_one_in_the_ProjectJson_when_one_is_set()
+        {
+            var json = new JObject();
+            json.Add("version", "1.1");
+            var project = GetProject(json);
+
+            project.Version.Should().Be(new NuGetVersion("1.1"));
+        }
+
+        [Fact]
+        public void It_sets_AssemblyFileVersion_to_the_ProjectJson_version_when_AssemblyFileVersion_is_no_passed_in_the_settings()
+        {
+            var json = new JObject();
+            json.Add("version", "1.1");
+            var project = GetProject(json);
+
+            project.AssemblyFileVersion.Should().Be(new NuGetVersion("1.1").Version);
+        }
+
+        [Fact]
+        public void It_sets_AssemblyFileVersion_Revision_to_the_AssemblyFileVersion_passed_in_the_settings_and_everything_else_to_the_projectJson_Version()
+        {
+            const int revision = 1;
+            var json = new JObject();
+            json.Add("version", "1.1");
+            var project = GetProject(json, new ProjectReaderSettings { AssemblyFileVersion = revision.ToString() });
+
+            var version = new NuGetVersion("1.1").Version;
+            project.AssemblyFileVersion.Should().Be(
+                new Version(version.Major, version.Minor, version.Build, revision));
+        }
+
+        [Fact]
+        public void It_throws_a_FormatException_when_AssemblyFileVersion_passed_in_the_settings_is_invalid()
```

Comment:
We should have a test for AssemblyFileVersion being a dot-seperated version number. I expect this will be common. Perhaps also a NuGet-style SemVer?